tests/drivers/gpio/gpio_enable_disable_interrupt/src/main.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_emul.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#define DEV    DT_GPIO_CTLR(DT_INST(0, test_gpio_enable_disable_interrupt), in_gpios)
+#define PIN_IN DT_GPIO_PIN(DT_INST(0, test_gpio_enable_disable_interrupt), in_gpios)
+
+static struct gpio_callback cb_data;
+static bool cb_called;
+
+static void callback(const struct device *dev, struct gpio_callback *gpio_cb, uint32_t pins)
+{
+	cb_called = true;
+}
+
+struct gpio_enable_disable_interrupt_fixture {
+	const struct device *port;
+	gpio_pin_t pin;
```

Comment:
You can use `struct gpio_dt_spec` here instead and then call `gpio_pin_configure_dt()` and `gpio_pin_interrupt_configure_dt()`.  

@cfriedt is it preferred for applications/samples to use `gpio_dt_spec` in place of separate `struct device` and `gpio_pin_t` fields?

Reply:
I updated this in another commit, if @cfriedt is also ok with that, then I will squash that commit.

Reply:
@cfriedt will be unavailable for a week or so.  But I think using `gpio_dt_spec` results in a cleaner implementation, so I think you should squash the changes.